index.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+'use strict';
+
+/**
+ * CleanKill hooks the interrupt handler, and provides callbacks for your code
+ * to cleanly shut down before the process exits.
+ */
+
+/**
+ * The type of a cleankill interrupt handler.
+ */
+export type Handler = (doneHandling: () => void) => void;
+
+let interruptHandlers: Handler[] = [];
+
+/**
+ * Register a handler to occur on SIGINT. All handlers are passed a callback,
+ * and the process will be terminated once all handlers complete.
+ */
+export function onInterrupt(handler: Handler): void {
+  interruptHandlers.push(handler);
+}
+
+/**
+ * Call all interrupt handlers, and call the callback when they all complete.
+ *
+ * Removes the list of interrupt handlers.
+ */
+export function close(done: () => void): void {
```

Comment:
We're moving to supporting only node ≥4 for our libraries. Converting to promises will be a major release though.